packages/data-service/src/data-service.ts

```diff
@@ -2645,6 +2695,31 @@ class DataServiceImpl extends WithLogContext implements DataService {
     };
   }
 
+  async listStreamProcessors(filter?: Document): Promise<StreamProcessor[]> {
+    const { streamProcessors } = await this._runStreamCmd({
+      listStreamProcessors: 1,
+      filter,
+    });
+    return streamProcessors;
+  }
+
+  async startStreamProcessor(name: string): Promise<void> {
+    await this._runStreamCmd({ startStreamProcessor: name });
+  }
+
+  async stopStreamProcessor(name: string): Promise<void> {
+    await this._runStreamCmd({ stopStreamProcessor: name });
+  }
+
+  async dropStreamProcessor(name: string): Promise<void> {
+    await this._runStreamCmd({ dropStreamProcessor: name });
+  }
+
+  private async _runStreamCmd(cmd: Document, options: Document = {}) {
+    const result = await this._database('admin', 'CRUD').command(cmd, options);
```

Comment:
We purposefully only use `runCommand` method when running commands directly against driver for the reasons mentioned in the method description. We probably should use the same approach here, or is it a deliberate decision not to?

Reply:
Have changed to use `runCommand`.